docs/01-app/03-api-reference/03-file-conventions/01-metadata/opengraph-image.mdx

```diff
@@ -402,15 +402,75 @@ export default async function Image({ params }) {
 
 #### Using Node.js runtime with local assets
 
-This example uses the Node.js runtime to fetch a local image on the file system and passes it as an `ArrayBuffer` to the `src` attribute of an `<img>` element. The local asset should be placed relative to the root of your project, rather than the location of the example source file.
+This example reads a local image from the file system and renders it in an Open Graph image.
+
+> **Path note:** Place your asset relative to the **project root** (e.g. `logo.png` next to `app/`). If you keep assets in `public/`, change the path to `join(process.cwd(), 'public', 'logo.png')`.
+
+**A) Base64 (TypeScript-friendly, recommended)**
+
+```tsx filename="app/opengraph-image.tsx" switcher
+import { ImageResponse } from 'next/og'
+import { join } from 'node:path'
+import { readFile } from 'node:fs/promises'
+
+export default async function Image() {
+  // If your file is in /public, use: join(process.cwd(), 'public', 'logo.png')
+  const logoData = await readFile(join(process.cwd(), 'logo.png'))
+  const logoSrc = `data:image/png;base64,${logoData.toString('base64')}`
+
+  return new ImageResponse(
+    (
+      <div
+        style={{
+          display: 'flex',
+          alignItems: 'center',
+          justifyContent: 'center',
+        }}
+      >
+        <img src={logoSrc} height="100" />
+      </div>
+    )
+  )
+}
+```
+
+```jsx filename="app/opengraph-image.js" switcher
+import { ImageResponse } from 'next/og'
+import { join } from 'node:path'
+import { readFile } from 'node:fs/promises'
+
+export default async function Image() {
+  // If your file is in /public, use: join(process.cwd(), 'public', 'logo.png')
+  const logoData = await readFile(join(process.cwd(), 'logo.png'))
+  const logoSrc = `data:image/png;base64,${logoData.toString('base64')}`
+
+  return new ImageResponse(
+    (
+      <div
+        style={{
+          display: 'flex',
+          alignItems: 'center',
+          justifyContent: 'center',
+        }}
+      >
+        <img src={logoSrc} height="100" />
+      </div>
+    )
+  )
+}
+```
+
+**B) ArrayBuffer / Typed Array (runtime-supported; TS suppression needed)**
```

Comment:
```suggestion
Passing an `ArrayBuffer` to the `src` attribute of an `<img>` element is not part of the HTML spec. The rendering engine used by `next/og` supports it, but because TypeScript definitions follow the spec, you need a `@ts-expect-error` directive or similar to use this [feature](https://github.com/vercel/satori/issues/606#issuecomment-2144000453).
```